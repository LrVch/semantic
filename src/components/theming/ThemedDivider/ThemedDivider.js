import {
  Divider
} from 'semantic-ui-react'
import styled from 'styled-components'

export default styled(Divider)`
  &&&:not(.vertical):not(.horizontal) {
    border-top-color:  ${({ theme }) => theme.second};
  }
`