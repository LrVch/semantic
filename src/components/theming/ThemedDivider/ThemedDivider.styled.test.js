import 'jest-styled-components'

import React from 'react'
import ThemedDivider from './ThemedDivider.js'
import renderer from 'react-test-renderer'
import styled from 'styled-components'

describe('ThemedDivider', () => {

  test('should create', () => {
    const tree = renderer.create(<ThemedDivider />).toJSON()

    expect(tree).toBeTruthy()
  })

  test('should has "border-top-color" equals to "fakeColor" ', () => {
    const tree = renderer.create(<ThemedDivider theme={{second: 'fakeColor'}} />).toJSON()

    expect(tree).toMatchSnapshot()
    expect(tree).toMatchSnapshot('border-top-color', 'fakeColor', {
      modifier: '&&&:not(.vertical):not(.horizontal)'
    })
  })
})


