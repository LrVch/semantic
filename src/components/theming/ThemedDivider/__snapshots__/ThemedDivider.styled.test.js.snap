// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemedDivider should has "border-top-color" equals to "fakeColor"  1`] = `
.c0.c0.c0:not(.vertical):not(.horizontal) {
  border-top-color: fakeColor;
}

<div
  className="ui divider c0"
  theme={
    Object {
      "second": "fakeColor",
    }
  }
/>
`;
